WWTMVC5/WWTWeb/MartianTile.aspx

```diff
@@ -9,76 +9,64 @@
 <%@ Import Namespace="System.Net" %>
 <%@ Import Namespace="WWTWebservices" %>
 <%
+ string query = Request.Params["Q"];
 
-    string query = Request.Params["Q"];
-    string[] values = query.Split(',');   
+
+ string[] values = query.Split(',');   
     int level = Convert.ToInt32(values[0]);
     int tileX = Convert.ToInt32(values[1]);
     int tileY = Convert.ToInt32(values[2]);
     string dataset = values[3];
-    string id = "nothing";
-	
-  	string DSSTileCache = WWTUtil.GetCurrentConfigShare("DSSTileCache", true);
 
 	switch(dataset)
 	{
 		case "mars_base_map":
-			id = "1738422189";
+			if (level < 18)
+    {
+       // Response.ContentType = "image/png";
+       
+        Stream s = PlateFile2.GetFileStream(Path.Combine(ConfigurationManager.AppSettings["WWTTilesDir"], "marsbasemap.plate"), -1, level, tileX, tileY);
+        
+        if (s == null || (int)s.Length == 0)
+        {
+            Response.Clear();
+            Response.ContentType = "text/plain";
+            Response.Write("No image");
+            Response.End();
+            return;
+        }
+        int length = (int)s.Length;
+        byte[] data = new byte[length];
+        s.Read(data, 0, length);
+        Response.OutputStream.Write(data, 0, length);
+        Response.Flush();
+        Response.End();
+       	 return;
+    	}
 			break;
 		case "mars_terrain_color":
-			id = "220581050";
-			break;
-		case "mars_hirise":
-			id = "109459728";
-			break;
-		case "mars_moc":
-			id = "252927426";
-			break;
-		case "mars_historic_green":
-			id = "1194136815";
-			break;
-		case "mars_historic_schiaparelli":
-			id = "1113282550";
-			break;
-		case "mars_historic_lowell":
-			id = "675790761";
-			break;
-		case "mars_historic_antoniadi":
-			id = "1648157275";
-			break;
-		case "mars_historic_mec1":
-			id = "2141096698";
-			break;
```

Comment:
If I'm reading this correctly, we're losing support for most of the other Mars datasets here. They're not heavily used so this is OK in the interim but this the kind of thing we should write down and make sure to patch up later.